gino/declarative.py

```diff
@@ -90,23 +97,11 @@ def __new__(mcs, name, bases, namespace, **new_kwargs):
                     args = table_args
 
             table = updates['__table__'] = Table(
-                table_name, mcs.metadata, *columns, *args, **table_kw)
-            namespace.update(updates)
-        rv = type.__new__(mcs, name, bases, namespace)
-        if table is not None:
-            table.__model__ = rv
-        return rv
-
-
-class Model(metaclass=ModelType):
-    __metadata__ = None
-    __table__ = None
-    query = Query()
-    update = Update()
-    delete = Delete()
-
-    def __init__(self, **values):
-        self.__values__ = values
+                table_name, cls.__metadata__, *columns, *args, **table_kw)
+            for k, v in updates.items():
+                setattr(cls, k, v)
+            table.__model__ = weakref.ref(cls)
```

Comment:
Should also change the usage of `__model__` to `__model__()` in `guess_model`. Would it be necessary to also change `query.__model__` to use `weakref`?